### PR TITLE
Add option for label formatting

### PR DESCRIPTION
**Pull request by sgal**
**What**:
New option for `babel-plugin-emotion` that allows to configure label format in generated class names. Implements https://github.com/emotion-js/emotion/issues/563

**Why**:
Details are in https://github.com/emotion-js/emotion/issues/563, but in general when debugging it is nice to have the file name in the class names (especially, for composed styles). The potential of the format is quite big. The feature and solution itself are inspired by `localIdentName` option in [webpack's css-loader](https://github.com/webpack-contrib/css-loader).

**How**:
New function that handles label generation for all cases (autoLabel, labelFormat, nothing).

**Checklist**:

- [x] Documentation
- [x] Tests
- [x] Code complete

### CODE REVIEW

**Comment by tkh44** [comment]
Thank you for your contribution! I took a quick look and it looks fine, but I'd like to have @mitchellhamilton look at it as well.

**Comment by sgal** [comment]
@tkh44 Thanks for the feedback! I'm currently thinking about how to handle cases when label format does not include `[local]` cause it might result in colliding class names.

One way would be to print warning and append `[local]` value automatically.
Another way is to throw an error with explanation that `[local]` is mandatory. But something in this one rubs me the wrong way.

What do you think?

**Comment by mitchellhamilton** [comment]
@sgal Collisions won't matter since there will be a hash as well and if the hash is the same then the style will be the same.

File: packages/babel-plugin-emotion/README.md [hunk 1]
```diff
@@ -197,11 +198,46 @@ const brownStyles = css({ color: 'brown' })
...
+const brownStyles = /*#__PURE__*/ css({ color: 'brown' }, 'label:brownStyles;')
 ```
 
 `brownStyles`'s value would be `css-1q8eu9e-brownStyles`
 
+### `labelFormat`
+
+`string`, defaults to `null`.
+
+This option automatically adds the `label` property (same as `autoLabel`), but allows to
```
Comment (mitchellhamilton) [inline packages/babel-plugin-emotion/README.md]: Should "but allows to" be "but allows you to"?

**Comment by sgal** [comment]
@mitchellhamilton Ah, thanks for clarifying the hashing. I fixed the phrasing in the docs.

**Comment by mitchellhamilton** [comment]
I've been thinking about this more and maybe the `labelFormat` option should default to `[local]` and only insert labels when the `autoLabel` option is true. So rather than having two options that enable labels where one of them also lets you configure the labels, just one to enable labels and one to configure the labels.

**Comment by sgal** [comment]
@mitchellhamilton That does sound reasonable and I think it is easier to understand as well. I'll update the PR tonight.

**Comment by matpaul** [comment]
@sgal, Hi! maybe also add path/folder or getLocalIdent function? In our project we use component/index.js, so filename - doesn't help here

**Comment by sgal** [comment]
@matpaul Which arguments do you think `getLocalIdent` function should take?

**Comment by matpaul** [comment]
@sgal 
full path to file (from project root) and local for example ?

**Pull request merged**